sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/runtime/MicroBatchExecution.scala

```diff
@@ -333,6 +375,29 @@ class MicroBatchExecution(
 
   private def initializeExecution(
       sparkSessionForStream: SparkSession): MicroBatchExecutionContext = {
+    var latestStartedBatch = offsetLog.getLatest()
+    val latestCommittedBatch = commitLog.getLatest()
+
+    val lastCommittedBatchId = latestCommittedBatch match {
+      case Some((batchId, _)) => batchId
+      case _ => -1L
+    }
+
+    // For a query running in Real-time Mode that fails after
+    // writing to offset log but before writing to commit log, we delete the extra
+    // entries in offsetLog to sync up. Note that this also means async checkpoint rollback handling
+    // is not compatible with low latency mode at this stage.
```

Comment:
low latency mode -> Real-time Mode. It should be better to keep consistent term in the doc.